### DISCUSSION TITLE
Filter by count in timeseries query

### DISCUSSION THREAD

**Discussion by GWeidmann** [datalust/seq-tickets#1672] [Help]
I have a query like this that shows me how much logs for a document come in at a given time

![image](https://user-images.githubusercontent.com/97590580/196641384-9ef589bb-e683-4b65-9765-7279126ade07.png)

I would like to filter out the ones that only have one or few entries to just focus on the ones with a lot of entries.

Unfortunately it's not possible to include a `having count > 3` in my query as seq doesn't allow it. Is there another possibility to make this query?

**Comment by nblumhardt** [comment]
Hi @GWeidmann, thanks for reaching out.

If you reverse the order of your groups, i.e. `Group by time(30m), Name`, then you can add `Having count > 3`. Unfortunately you won't be able to chart the result, but the results should match what you're after.

Hope this helps,
Nick

**Reply by GWeidmann** [reply]
Is this a planned feature to allow this? It would be quite helpful.

**Reply by nblumhardt** [reply]
There's nothing specific planned for this currently; I think there are two possible directions to explore:

 * Disabling/rethinking the default "fill" that's applied to timeseries results (`time()` in the last group position). This is the reason for the restriction on `time()` groupings and `having` clauses, where fill (e.g. the zero line points you'll see in a grouped `count(*)` for time slices in which that group wasn't present) is applied after group filtering, and so isn't affected by `having` clauses. Without applying fill, `having` clauses could be processed normally - though this isn't desirable in all cases. Needs some more analysis.
 * Or, making the UI able to chart data sets where each time slice carries a set of grouped rows (with groups that may differ from other time slices); maybe a more direct/less disruptive option, but also still quite a bit to figure out.

Great to have a concrete use case for us to consider; if I come up with any other options/workarounds, I'll loop back here. Thanks again!
